vue-components/src/styles/mixins/Typography.scss

```diff
@@ -0,0 +1,80 @@
+@mixin heading-1 ($color: $font-color-emphasized) {
+    font-family: $font-family-style-heading-serif;
+    font-size: $font-size-style-h1;
+    font-weight: $font-weight-style-h1;
+    line-height: $font-line-height-style-heading;
+    color: $color;
+}
+
+@mixin heading-2 ($color: $font-color-emphasized) {
+    font-family: $font-family-style-heading-sans;
+    font-size: $font-size-style-h2;
+    font-weight: $font-weight-style-h2;
+    line-height: $font-line-height-style-heading;
+    color: $color;
+}
+
+@mixin heading-3 ($color: $font-color-emphasized) {
+    font-family: $font-family-style-heading-sans;
+    font-size: $font-size-style-h3;
+    font-weight: $font-weight-style-h3;
+    line-height: $font-line-height-style-heading;
+    color: $color;
+}
+
+@mixin heading-4 ($color: $font-color-emphasized) {
+    font-family: $font-family-style-heading-sans;
+    font-size: $font-size-style-h4;
+    font-weight: $font-weight-style-h4;
+    line-height: $font-line-height-style-heading;
+    color: $color;
+}
+
+@mixin heading-5 ($color: $font-color-emphasized) {
+    font-family: $font-family-style-heading-sans;
+    font-size: $font-size-style-h5;
+    font-weight: $font-weight-style-h5;
+    line-height: $font-line-height-style-heading;
+    color: $color;
+}
+
+@mixin heading-6 ($color: $font-color-emphasized) {
+    font-family: $font-family-style-heading-sans;
+    font-size: $font-size-style-h6;
+    font-weight: $font-weight-style-h6;
+    line-height: $font-line-height-style-heading;
+    color: $color;
+}
+
+@mixin body ($color: $font-color-base, $line-height: $font-line-height-style-body, $font-weight: $font-weight-style-body) {
+    font-family: $font-family-style-body;
+    font-size: $font-size-style-body;
+    font-weight: $font-weight; //can also be $font-weight-style-body-bold
```

Comment:
here's the clumsiness! is there a more elegant way of allowing implementers to select a set of variables as values for these properties?